### ISSUE TITLE
find `source` on `$PATH`

### ISSUE THREAD

**Issue by sellout** [koalaman/shellcheck#2668]
#### For bugs
- Rule Id (if any, e.g. SC1091): 
- My shellcheck version (`shellcheck --version` or "online"): 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:
strict-mode.bash (on `$PATH`)
```sh
set -e
```
my script
```sh
#/usr/bin/env bash
source strict-mode.bash

cd some/path
cat foo
```
#### Here's what shellcheck currently says:
```
SC2164 (warning): Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
```
#### Here's what I wanted or expected to see:
No warning.

If I inline `set -e` into my script, I get no warning about `cd`.

Bash finds “bare” sourced files on `$PATH`, so Shellcheck should check the `$PATH` in this case.

**Comment by sellout** [comment]
Sorry, I had a lot of interruptions while writing this. The thing I wanted to point out was that using `source=/dev/null` isn’t sufficient, because the contents of strict-mode.bash affect how the rest of the script is interpreted (SC2164).

And also to ask if there’s any workaround for this currently – I don’t have a static path to point to for the location of strict-mode.bash.

**Comment by sellout** [comment]
BTW, I don’t think I _can_ try this on shellcheck.net, as I don’t know what shell scripts (if any) might exist on the `PATH` (if any) there.

**Comment by port19x** [comment]
If you have no static path for strict mode thats tough luck
[SC1090](https://github.com/koalaman/shellcheck/wiki/SC1090)

---

The issue is that shellcheck only sources from absolute paths.
You also need `-x` to check sourced files.

Given a `my-script.bash` like the following
```sh
#!/usr/bin/env bash
source /usr/local/bin/strict-mode.bash

cd some/path
cat foo
```

`shellcheck -x my-script.bash` no longer complains

---

A workaround could be to, instead of sourcing it, just call strict-mode, tho I don't think that has the desired effect.

So I guess I can reproduce your issue, tho I'm not sure if something can or should be done about it
